Applications/Scale/test/testScale.cpp

```diff
@@ -62,30 +62,80 @@ int main()
-        const SimTK::Vec3& location = marker.get_location();
+    // Check number of Marker Components (otherwise, test will pass even if
+    // markers are missing from result model).
+    {
+        const unsigned nmResult = result.countNumComponents<Marker>();
+        const unsigned nmTarget = target->countNumComponents<Marker>();
+        OPENSIM_THROW_IF(nmResult != nmTarget, Exception,
+            "Incorrect number of Marker Components in result Model.");
+    }
+
+    // Check Marker locations.
+    cout << "Checking marker locations..." << endl;
+    for (const Marker& mResult : result.getComponentList<Marker>())
+    {
+        const std::string& path = mResult.getAbsolutePathString();
 
-        std::cout << "Marker `" << name << "' location: " << location
-            << " | " << stdLocation << " (standard) " << endl;
+        // Ensure marker exists in target model.
+        OPENSIM_THROW_IF(!target->hasComponent(path), Exception,
+            "Marker '" + path + "' not found in standard model.");
 
-        ASSERT_EQUAL(stdLocation, location, tol, __FILE__, __LINE__,
-            "Scale model marker location failed to match the standard");
+        const Vec3& result_loc = mResult.get_location();
+        const Vec3& target_loc = target->getComponent<Marker>(path).get_location();
+
+        cout << "  '" << path << "' - location: " << result_loc << endl;
+        ASSERT_EQUAL(result_loc, target_loc, tol, __FILE__, __LINE__,
+            "Marker '" + path + "' location in scaled model does not match "
+            + "standard of " + target_loc.toString());
+    }
+
+    // Check number of GeometryPath Components (otherwise, test will pass even
+    // if path actuators, ligaments, etc. are missing from result model).
+    {
+        const unsigned ngpResult = result.countNumComponents<GeometryPath>();
+        const unsigned ngpTarget = target->countNumComponents<GeometryPath>();
+        OPENSIM_THROW_IF(ngpResult != ngpTarget, Exception,
+            "Incorrect number of GeometryPath Components in result Model.");
+    }
+
+    // Check GeometryPath path point locations.
+    cout << "Checking path point locations..." << endl;
+    for (const GeometryPath& gpResult : result.getComponentList<GeometryPath>())
+    {
+        const std::string& path = gpResult.getAbsolutePathString();
+
+        // Ensure GeometryPath exists in target model.
+        OPENSIM_THROW_IF(!target->hasComponent(path), Exception,
+            "GeometryPath '" + path + "' not found in standard model.");
+
+        cout << "  '" << path << "'" << endl;
+        for (int i = 0; i < gpResult.getPathPointSet().getSize(); ++i)
+        {
+            const Vec3& result_loc = static_cast<const PathPoint&>(
+                gpResult.getPathPointSet()[i]).get_location();
+            const Vec3& target_loc = static_cast<const PathPoint&>(
+                target->getComponent<GeometryPath>(path).getPathPointSet()[i])
+                .get_location();
```

Comment:
I wonder if the Appveyor failure would go away if you use `getLocation(s)` instead of `get_location()`?

Reply:
Thanks. Yes, I discussed with @aseth1. I shouldn't be `static_cast`ing because the `PathPointSet` can contain `MovingPathPoint`s which don't have a `location` property.